ui-modules/blueprint-composer/app/components/util/d3-blueprint.js

```diff
@@ -771,50 +774,37 @@ export function D3Blueprint(container, options) {
         return node;
     }
 
+    /**
+     * Draws new relationships as arcs and labels. Updates existing ones.
+     */
     function drawRelationships() {
+
         showRelationships();
 
-        let relationData = _relationGroup.selectAll('.relation')
-            .data(_d3DataHolder.visible.relationships, (d)=>(d.source._id + '_related_to_' + d.target._id));
+        const getPathId = (d) => (d.source._id + '-' + d.target._id);
+        const getRelationId = (d) => (getPathId(d) + '-' + d.pathSelector);
+
+        // ====== RELATIONSHIP ARCS ===========
 
-        let relationDataEntered = relationData.enter();
+        let arcsData = Object.values(_d3DataHolder.visible.relationships.reduce((accumulator, d) => {
```

Comment:
Due to only pulling `Object.values`, in the end `arcsData` will be an array of all the `{source,target,pathSelector}` objects it seems. If so, it might be easier to just do

```
const arcsData = _d3DataHolder.visible.relationships;
// or in case it's important to retain specific properties only
const arcsData = _d3DataHolder.visible.relationships
  .map(({ source, target, pathSelector }) => ({ source, target, pathSelector }));
```

Reply:
There is a need to reduce, not to map. See code comments.